### PR TITLE
CI and pkgdown improvements

### PR DESCRIPTION
**Pull request by cpsievert**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## Pull Request
+
+Before you submit a pull request, please ensure you've completed the following checklist
+
+- [ ] Ensure there is an already open and relevant [GitHub issue](https://github.com/rstudio/bootstraplib/issues/new) describing the problem in detail and you've already received some indication from the maintainers that they are welcome to a contribution to fix the problem. This helps us to prevent wasting anyone's time. 
+
+- [ ] Ensure that you have signed the [individual](https://rstudioblog.files.wordpress.com/2017/05/rstudio_individual_contributor_agreement.pdf) or [corporate](https://rstudioblog.files.wordpress.com/2017/05/rstudio_corporate_contributor_agreement.pdf) contributor agreement as appropriate. You can send the signed copy to jj@rstudio.com.
```
Comment (schloerke) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: Ask Jeff for CLA permission to the repo and remove this.

File: .github/workflows/R-CMD-check.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: macOS-latest,   r: 'release'}
+          - {os: windows-latest, r: 'release'}
+          - {os: windows-latest, r: '3.6'}
+          - {os: ubuntu-16.04,   r: 'devel', rspm: "https://packagemanager.rstudio.com/cran/__linux__/xenial/latest", http-user-agent: "R/4.0.0 (ubuntu-16.04) R (4.0.0 x86_64-pc-linux-gnu x86_64 linux-gnu) on GitHub Actions" }
```
Comment (schloerke) [inline .github/workflows/R-CMD-check.yaml:30]: I know the default is ubuntu 16... why not 18 / bionic to help avoid updating when 16 is removed in the far future?

File: .github/workflows/pkgdown.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        with:
+          r-version: release
+
+      - uses: r-lib/actions/setup-pandoc@master
+
+      - name: git setup
+        shell: bash
+        run: |
+          git config --global user.email "carson@rstudio.com"
+          git config --global user.name "Carson Sievert"
```
Comment (schloerke) [inline .github/workflows/pkgdown.yaml:28]:
```suggestion
```

File: .github/workflows/pkgdown.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Install package
+        run: R CMD INSTALL .
+
+      - name: Build Site (PR)
+        if: github.event_name != 'push'
+        run: pkgdown::build_site()
+        shell: Rscript {0}
+
+      - name: Build and Deploy Site
```
Comment (schloerke) [inline .github/workflows/pkgdown.yaml:58]:
```suggestion
      - name: git setup
        if: github.event_name == 'push'
        run: |
          git config --local user.email "${GITHUB_ACTOR}@users.noreply.github.com" && \
            git config --local user.name "${GITHUB_ACTOR}"

      - name: Build and Deploy Site
```

File: .github/workflows/pkgdown.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+          remotes::install_deps(dependencies = TRUE)
+        shell: Rscript {0}
+
+      - name: Install package
+        run: R CMD INSTALL .
+
+      - name: Build Site (PR)
+        if: github.event_name != 'push'
+        run: pkgdown::build_site()
+        shell: Rscript {0}
```
Comment (schloerke) [inline .github/workflows/pkgdown.yaml:56]:
```suggestion
      - name: Build Site (PR)
        if: github.event_name != 'push'
        run: pkgdown::build_site()
        shell: Rscript {0}

      - name: Validate all topics exist (PR)
        if: github.event_name != 'push'
        shell: Rscript {0}
        run: |
          pkgdown::build_reference_index()
          stopifnot(length(warnings()) == 0)
```

File: .github/workflows/R-CMD-check.yaml (hunk 2 above)
Comment (cpsievert) [reply .github/workflows/R-CMD-check.yaml:30]: Yea, Jeroen and Jim had an interesting back-and-forth on #r-lib about this the other day (Jim's in the 16.04 all the way camp). In thematic, I recently changed devel to 20.04, and release to 18.04...I'd be game for doing the same here

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (cpsievert) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Oh it's actually already functional

File: .github/workflows/R-CMD-check.yaml (hunk 2 above)
Comment (schloerke) [reply .github/workflows/R-CMD-check.yaml:30]: Up to you which you choose.  Just make sure to use the `matrix.config.os` value when installing dependencies.  Also be sure to use the `matrix.config.os` value in your cache path

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (schloerke) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: woot. Just need to remove then. 😄

**Pull request merged**